Sources/Tracing/Docs.docc/Guides/InstrumentYourLibrary.md

```diff
@@ -0,0 +1,294 @@
+    func inject(_ value: String, forKey key: String, into request: inout HTTPRequest) {
+        request.headers.append((key, value))
+    }
+}
+```
+
+Once the metadat has been injected, the request–including all the additional metadata–is sent over the network.
+
+> Note: The actual logic of deciding what baggage values to inject depend on the tracer implementation, and thus we are not covering it in this _end-user_ focused guide. Refer to <doc:ImplementATracer> if you'd like to learn about implementing a ``TracerProtocol``.
+
+#### Handling inbound requests
+
+On the receiving side, an HTTP server needs to perform the inverse operation, as it receives the request from the network and forms an `HTTPRequest` object, and before it passed it to user-code, it must extract any trace metadata from the request headers into the `Baggage`. 
+
+```
+                   ┌──────────────────────────────────┐
+                   │   Specific Tracer / Instrument   │
+                   └──────────────────────────────────┘
+                                      ▲
+                                      │
+      instrument.extract(request, into: &baggage, using: httpExtractor)       
+                                      │                                  
+                               ┌──────▼──────┐                           
+┌────┐          ┌──────────────┤ Tracing API ├────────────┐              
+│ N  │          │HTTPServerLib └──────▲──────┘            │              
+│ e  │          │                     │                   │              
+│ t  │          │┌───────────────┐  ┌─▼───────────────┐   │              
+│ w  │          ││     parse     │  │extract metadata │   │     ┌───────────┐
+│ o  │─────────▶││  HTTPRequest  ├──▶from HTTPRequest │───┼────▶│ User code │
+│ r  │          │└───────────────┘  └─────────────────┘   │     └───────────┘
+│ k  │          │                                         │              
+│    │          └─────────────────────────────────────────┘              
+└────
+```
+
+This is very similar to what we were doing on the outbound side, but the roles of baggage and request are somewhat inversed: we're extracting values from the carrier into the baggage. The code performing this task could look something like this:
+
+```swift
+func handler(request: HTTPRequest) async {
+  // we are beginning a new "top level" context - the beginning of a request - 
+  // and thus start from a fresh, empty, top-level baggage:
+  var baggage = Baggage.topLevel
+
+  // populate the baggage by extracting interesting metadata from the incoming request:
+  InstrumentationSystem.instrument.extract(
+    request,
+    into: &baggage,
+    using: HTTPRequestExtractor()
+  )
+
+  // ... invoke user code ...
+}
+```
+
+Similarily to the outbound side, we need to implement an ``Extractor`` because the tracing libraries don't know about our specific HTTP types, yet we need to have them decide for which values to extract keys.
```

Comment:
```suggestion
Similarly to the outbound side, we need to implement an ``Extractor`` because the tracing libraries don't know about our specific HTTP types, yet we need to have them decide for which values to extract keys.
```

Reply:
Or even Silmarillion-y 😂 Man, this word is my arch-enemy... thanks for spotting it